### PR TITLE
Check that token is AsyncKeyword before calling lookAhead

### PR DESCRIPTION
**Pull request by yuit**
Before calling lookAhead function check that the current toke is AsyncKeyword because lookAHead is an expensive operation

### CODE REVIEW

**Comment by RyanCavanaugh** [comment]
Is there a bug tracking this?

**Comment by yuit** [comment]
@RyanCavanaugh nop it is from the [PR](https://github.com/Microsoft/TypeScript/pull/8444) @mhegazy made a comment after I merged the PR in

File: src/compiler/parser.ts [hunk 1]
```diff
@@ -2989,11 +2989,14 @@ namespace ts {
...
 
         function tryParseAsyncSimpleArrowFunctionExpression(): ArrowFunction {
-            const isUnParenthesizedAsyncArrowFunction = lookAhead(isUnParenthesizedAsyncArrowFunctionWorker);
-            if (isUnParenthesizedAsyncArrowFunction === Tristate.True) {
-                const asyncModifier = parseModifiersForArrowFunction();
-                const expr = parseBinaryExpressionOrHigher(/*precedence*/ 0);
-                return parseSimpleArrowFunctionExpression(<Identifier>expr, asyncModifier);
+            // We do a check here so that we won't be doing unnecessarily call to "lookAhead"
+            if (token === SyntaxKind.AsyncKeyword) {
+                const isUnParenthesizedAsyncArrowFunction = lookAhead(isUnParenthesizedAsyncArrowFunctionWorker);
```
Comment (RyanCavanaugh) [inline src/compiler/parser.ts]: Linter's mad about the trailing WS here

**Comment by sandersn** [comment]
:+1:

**Pull request merged**